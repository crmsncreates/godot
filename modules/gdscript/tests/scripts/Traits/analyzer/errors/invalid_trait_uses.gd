trait TraitA extends Node:
	pass

class ClassA:
	pass

class ClassB extends Object:
	uses TraitA # Using a trait that extends Node but the class extends Object.
    # This should raise an error because inheritance mismatch.
    # in this case, Node or children classes of Node would be accepted as using class inheritance.

class ClassC:
	uses ClassA # Class cannot be used as a trait. Should raise an error.
