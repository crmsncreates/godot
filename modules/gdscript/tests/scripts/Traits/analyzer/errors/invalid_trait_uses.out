GDTEST_ANALYZER_ERROR
>> ERROR at line 8: "TraitA" trait extending "Node" does not match class inheritance.
>> ERROR at line 13: "ClassA" is not a trait.
